pkg/kubelet/volume/empty_dir/empty_dir.go

```diff
@@ -197,36 +201,51 @@ func (ed *emptyDir) GetPath() string {
 
 // TearDown simply discards everything in the directory.
 func (ed *emptyDir) TearDown() error {
-	switch ed.medium {
-	case api.StorageTypeDefault:
-		return ed.teardownDefault()
-	case api.StorageTypeMemory:
-		return ed.teardownTmpfs()
-	default:
-		return fmt.Errorf("unknown storage medium %q", ed.medium)
+	return ed.TearDownAt(ed.GetPath())
+}
+
+// TearDownAt simply discards everything in the directory.
+func (ed *emptyDir) TearDownAt(dir string) error {
+	// Figure out the medium.
+	if medium, err := ed.mediumer.GetMedium(dir); err != nil {
+		glog.Errorf("TIM: error from GetMedium: %v", err)
+		return err
+	} else {
+		switch medium {
+		case mediumMemory:
+			ed.medium = api.StorageTypeMemory
```

Comment:
yeah, and if it detect that, SetUp short circuits (idempotency matters!)
but TearDown does not (tries to unmount).

Looking at a fix now.

On Mon, Mar 16, 2015 at 2:50 PM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/kubelet/volume/empty_dir/empty_dir.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5167#discussion_r26531245>
> :
>
> > -		return ed.teardownTmpfs()
> > -	default:
> > -		return fmt.Errorf("unknown storage medium %q", ed.medium)
> > +	return ed.TearDownAt(ed.GetPath())
> > +}
> > +
> > +// TearDownAt simply discards everything in the directory.
> > +func (ed *emptyDir) TearDownAt(dir string) error {
> > +	// Figure out the medium.
> > +	if medium, err := ed.mediumer.GetMedium(dir); err != nil {
> > +		glog.Errorf("TIM: error from GetMedium: %v", err)
> > +		return err
> > +	} else {
> > +		switch medium {
> > +		case mediumMemory:
> > +			ed.medium = api.StorageTypeMemory
>
> On F20, any path in "/tmp" returns the same linuxTmpfsMagic number.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5167/files#r26531245>
> .
>